src/client/renderingengine.cpp

```diff
@@ -633,7 +633,8 @@ float RenderingEngine::getDisplayDensity()
 
 float RenderingEngine::getDisplayDensity()
 {
-	return (g_settings->getFloat("screen_dpi") / 96.0) * g_settings->getFloat("display_density_factor");
+	return (std::max(g_settings->getFloat("screen_dpi"), 1.0f) / 96.0) *
+		g_settings->getFloat("display_density_factor");
```

Comment:
`display_density_factor` needs limiting too, the end result may not be zero or negative
(might be easier to just check that once before returning)